test/testhelpers.jl

```diff
@@ -1,19 +1,20 @@
 using MPI
 
 function runmpi(file; ntasks = 1)
-    MPI.Initialized() &&
-    !MPI.Finalized() &&
-    error(
-        "runmpi does not work if MPI has been " *
-        "Initialized but not Finalized",
-    )
-
+    if MPI.Initialized() && !MPI.Finalized()
```

Comment:
should we just change this to error if `MPI.Initialized()`?

Reply:
that makes sense, I think you need both in the case of callbacks registered during finalization IIRC which we don't use.

Reply:
I don't really understand why this check is needed at all given that run test harness is run in a separate process, is there a reason this check is included?

Reply:
Not sure either: perhaps there were problems if you tried to launch an MPI process from within another one? 

We can probably get rid of it for now.